config/config.go

```diff
@@ -584,6 +584,10 @@ type DisabledMetrics struct {
 	// that were created or reused.
 	AdapterConnectionMetrics bool `mapstructure:"adapter_connections_metrics"`
 
+	// True if we don't want to collect metrics on the number of adapter
+	// connection dials and the time said dial takes
+	AdapterConnectionDialMetrics bool `mapstructure:"adapter_connection_dial_metrics"`
```

Comment:
Nitpick: Please update the description to make it clearer this is for dial time histograms and dial errors.